docs/src/main/asciidoc/rest-client-reactive.adoc

```diff
@@ -0,0 +1,594 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+= Quarkus - Using the REST Client Reactive
+
+include::./attributes.adoc[]
+
+This guide explains how to use the REST Client Reactive in order to interact with REST APIs.
+REST Client Reactive is a non-blocking counterpart of the RESTEasy REST Client.
+
+If you wish to both, use the client and expose REST endpoints from your application,
+it is advised to use link:resteasy-reactive.adoc[RESTEasy Reactive] as the server implementation.
```

Comment:
I think this needs to be worded more strongly, i.e when using RESTEasy Reactive this client is absolutely preferred over the legacy `quarkus-rest-client`